hazelcast/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -444,64 +445,137 @@ static ProcessorMetaSupplier forceTotalParallelismOne(
-                                    + "supports only total parallelism of 1. Local parallelism must be 1.");
-                }
+        @Override
+        public void init(@Nonnull Context context) throws Exception {
+            PermissionsUtil.checkPermission(supplier, context);
+            if (context.localParallelism() != 1) {
+                throw new IllegalArgumentException(
+                        "Local parallelism of " + context.localParallelism() + " was requested for a vertex that "
+                                + "supports only total parallelism of 1. Local parallelism must be 1.");
             }
+        }
 
-            @Nonnull @Override
-            public Function<? super Address, ? extends ProcessorSupplier> get(@Nonnull List<Address> addresses) {
-                if (!addresses.contains(memberAddress)) {
-                    throw new JetException("Cluster does not contain the required member: " + memberAddress);
-                }
-                return addr -> addr.equals(memberAddress) ? supplier : count -> singletonList(new ExpectNothingP());
+        @Nonnull @Override
+        public Function<? super Address, ? extends ProcessorSupplier> get(@Nonnull List<Address> addresses) {
+            if (!addresses.contains(memberAddress)) {
+                throw new JetException("Cluster does not contain the required member: " + memberAddress);
             }
+            return new AddressProcessorSupplierFunction(supplier, memberAddress);
+        }
 
-            @Override
-            public int preferredLocalParallelism() {
-                return 1;
-            }
+        @Override
+        public int preferredLocalParallelism() {
+            return 1;
+        }
 
-            @Override
-            public void writeData(ObjectDataOutput out) throws IOException {
-                out.writeObject(supplier);
-                out.writeObject(memberAddress);
-            }
+        @Override
+        public void writeData(ObjectDataOutput out) throws IOException {
+            out.writeObject(supplier);
+            out.writeObject(memberAddress);
+        }
 
-            @Override
-            public void readData(ObjectDataInput in) throws IOException {
-                supplier = in.readObject();
-                memberAddress = in.readObject();
-            }
+        @Override
+        public void readData(ObjectDataInput in) throws IOException {
+            supplier = in.readObject();
+            memberAddress = in.readObject();
         }
 
-        return new SpecificMemberPms(supplier, memberAddress);
+        @Override
+        public int getFactoryId() {
+            return JetDataSerializerHook.FACTORY_ID;
+        }
+
+        @Override
+        public int getClassId() {
+            return JetDataSerializerHook.SPECIFIC_MEMBER_PROCESSOR_META_SUPPLIER;
+        }
+    }
+
+    class AddressProcessorSupplierFunction implements Function<Address, ProcessorSupplier>, IdentifiedDataSerializable {
```

Comment:
This function is never serialized, no point in making it IDS.

Reply:
Fixed